commands/checkout_test.go

```diff
@@ -0,0 +1,16 @@
+package commands
+
+import (
+	"testing"
+
+	"github.com/github/hub/Godeps/_workspace/src/github.com/bmizerany/assert"
+)
+
+func TestReplaceCheckoutParam(t *testing.T) {
+	checkoutURL := "https://github.com/github/hub/pull/12"
+	args := NewArgs([]string{"checkout", "-b", checkoutURL})
+	replaceCheckoutParam(args, checkoutURL, "jingweno", "origin/master")
+
+	cmd := args.ToCmd()
+	assert.Equal(t, "git checkout -b --track -B jingweno origin/master", cmd.String())
```

Comment:
Not sure about this assertion. The original invocation is an invalid usage of the `PULLREQ-URL` parameter. Maybe we shouldn't transform the arguments in this case?